### ISSUE TITLE
Tesseract performs differently when used in local machine Ubuntu image and on other

### ISSUE THREAD

**Issue by Sultan91** [tesseract-ocr/tesseract#1254]
We have developed python based app with converts photo pictures into XML documents, parsed by key fields. The software was installed on Ubuntu virtual OS on local PC and works correctly. However, when porting the same Ubuntu image on other PC and running application from there it outputs recognized text in different manner (different words order). The problem is exactly at OCR Tesseract operation stage.

P.S: Also, we tried to install and run our software on ordinary PCs also Ubuntu and it shows also different OCR outputs.

### Environment

* **Tesseract Version**: 3.04.01
* **Platform**: Ubuntu 64bit

**Comment by amitdo** [comment]
My guess is that the issue occurs due to: 
https://www.google.co.il/search?q=floating+point+differences+between+machines
https://www.google.co.il/search?q=floating+point+determinism

**Comment by Sultan91** [comment]
Thanks for response, also I've noticed after running basic tesseract recognition on my PC and my friend's one it outputs different results. This is scary(

**Comment by stweil** [comment]
I would be surprised if FP calculations were the reason. As 3.04.01 is a rather old version, I suggest to try the experimental packages for 4.00 from Debian (https://packages.debian.org/sid/all/tesseract-ocr-eng/download).

**Comment by stweil** [comment]
I also run Tesseract on different computers, and it is very important for me that the OCR results are always the same. Up to now, I did not notice differences as long as I used the same software version, training data and program parameters.

Software bugs can be one reason for program results which are not reproducible (not even on the same computer), especially heap errors like use after free or code which uses undefined variables. Both Tesseract and software used by Tesseract like Leptonica had such bugs in the past (see for example commit fb863c97a9c8e9c05874af6f6fbd77923cc6f6e9). I don't know such bugs in the latest versions.
